### PR TITLE
Speed up the post for categorisation by limiting the select * from influx to the date period we have loaded

### PR DESCRIPTION
**Pull request by Hobbsee**

### CODE REVIEW

File: lib/influx.rb [hunk 1]
```diff
@@ -182,11 +182,23 @@ def noise_candidates(start_date = nil, end_date = nil)
...
-      data.each do |incident, category|
-        query = "select * from #{timeseries} where id = '#{incident}'"
-        current_point = @influxdb.query(query)[timeseries].first
+      oldest =  Chronic.parse(opts[:start_date], :guess => false).first.to_i
+      newest =  Chronic.parse(opts[:end_date], :guess => false).last.to_i
+      begin
+        entries = @influxdb.query "select * from #{timeseries} where time > #{oldest}s and time < #{newest}s"
+        entries = entries.empty? ? [] : entries[timeseries]
+      rescue InfluxDB::Error => e
+        if e.message.match(/^Couldn't find series/)
```
Comment (nerdyness) [inline lib/influx.rb]: How would you feel about this one-liner?

`if e.message.match(/^Couldn't find series/) ? return : raise`

What a weird bug to have?! :smile:

**Comment by nerdyness** [comment]
Merge at will, looks good :+1:

File: lib/influx.rb (hunk 1 above)
Comment (Hobbsee) [inline lib/influx.rb]: Theoretically, we should never get in this state - opts[:data] will always be empty if the series doesn't exist.  The reason this code is here is to mirror the code above (and to not explode if influx goes away between when the page was loaded and when the save button was hit)

However, if we do get here, should we still try writing the data on the point we have, or completely aborting?  My code currently does the former, and your edit is suggesting the latter.

Comment (nerdyness) [inline lib/influx.rb]: > However, if we do get here, should we still try writing the data on the point we have, or completely aborting? My code currently does the former, and your edit is suggesting the latter.

Yeah, it was a suggestion on my part and at the time of writing I thought the 2 bits of code were equivalent but they're not, oops. I agree trying to write is better than completely aborting.

File: pigeonhole.rb [hunk 2]
```diff
@@ -82,10 +82,15 @@
 
 post '/:start_date/:end_date' do
   uri = "#{params["start_date"]}/#{params["end_date"]}"
+  opts = {
+    :start_date => params[:"start_date"],
+    :end_date   => params[:end_date]
+  }
```
Comment (rebyn) [inline pigeonhole.rb]: Are these inconsistencies intentional? Also with `uri = "#{params["start_date"]}/#{params["end_date"]}"`.

**Pull request merged**